test/unit/bin-tessel-2.js

```diff
@@ -924,42 +922,84 @@ exports['Tessel (init)'] = {
     done();
   },
 
-  noOpts: function(test) {
-    test.expect(5);
-    // Call the CLI with the init command
+  'defaults to --lang=js': function(test) {
+    test.expect(3);
+
+    var resolve = Promise.resolve();
+    this.createNewProject = this.sandbox.stub(controller, 'createNewProject').returns(resolve);
+
     cli(['init']);
 
-    setImmediate(() => {
-      // Ensure it calls our internal init function
-      test.equal(this.initProject.callCount, 1);
-      // Ensure it checks the language being requested
-      test.equal(this.resolveLanguage.callCount, 1);
-      // It should generate a js project
-      test.equal(this.generateJavaScriptProject.callCount, 1);
-      // It should not generate a Rust project
-      test.equal(this.generateRustProject.callCount, 0);
-      // Ensure it continues to call our exit function
-      test.equal(this.successfulCommand.callCount, 1);
+    resolve.then(() => {
```

Comment:
All CLI operations result in a call to a function that returns a promise, therefore all must be tested by observing the outcome once resolved or rejected.